nwbwidgets/utils/dynamictable.py

```diff
@@ -1,21 +1,41 @@
 from pynwb.core import DynamicTable
+import numbers
 import numpy as np
+from typing import Iterable
 
 
-def infer_categorical_columns(dynamic_table: DynamicTable):
+def infer_categorical_columns(dynamic_table: DynamicTable, region: Iterable = None):
+    """
+    Parameters
+    ----------
+    dynamic_table: DynamicTable
+    region: Iterable
+        row indices to select
 
+    Returns
+    -------
+    categorical_cols: dict()
+        keys: as columns that are categorical, values as the unique values
+    """
     categorical_cols = {}
+    region = region if region is not None else list(range(len(dynamic_table)))
     for name in dynamic_table.colnames:
         if len(dynamic_table[name].shape) == 1:
-            try:  # TODO: fix this
-                unique_vals = np.unique(dynamic_table[name].data)
-                if 1 < len(unique_vals) <= (len(dynamic_table[name].data) / 2):
+            try:
+                if isinstance(dynamic_table[name].data[0], (str, numbers.Number)):
```

Comment:
should also include `bytes`

Reply:
oh yes correct. fixed here https://github.com/NeurodataWithoutBorders/nwb-jupyter-widgets/pull/157/commits/558b4077f29ea9ed593a3a4c0fb3f755328ad598